web/src/app/profile/approved-researcher/components/ApprovedResearcher.tsx

```diff
@@ -4,18 +4,39 @@ import { useAuth } from "../../../hooks/useAuth";
 import LoginFallback from "@/components/ui/LoginFallback";
 import ApprovedResearcherForm from "./ApprovedResearcherForm";
 import AgreementText from "./AgreementText";
+import { Agreement, getAgreementsApprovedResearcher } from "@/openapi";
+import { useEffect, useState } from "react";
 
 export default function ApprovedResearcher() {
-  const { loading, isAuthed } = useAuth();
+  const { isAuthed } = useAuth();
+  const [agreement, setAgreement] = useState<Agreement | null>(null);
 
-  if (loading) return null;
+  useEffect(() => {
+    const fetchAgreement = async () => {
+      try {
+        const res = await getAgreementsApprovedResearcher();
+
+        if (res.response.status === 200 && res.data) {
+          setAgreement(res.data);
+        }
+      } catch (err) {
+        console.error("Agreement fetch error:", err);
+      }
+    };
+
+    if (isAuthed) {
+      fetchAgreement();
+    }
+  }, [isAuthed]);
 
```

Comment:
This was embedded within the `AgreementsText.tsx` component before as I considered it to be specifically tied to fetching and displaying the text itself. This way is fine too though. The only other advantage of having it the way it was before is that you can assume the auth state has already been established which made the code slightly cleaner.

But these a small differences, happy for this approach to remain 👍

Reply:
Ah this actually makes more sense now as I see you are passing data to multiple components!